tests/settings/manager_test.py

```diff
@@ -136,10 +137,67 @@ def test_dont_choke_on_regex_special_chars_in_tagstring(self):
                         normal = '','', 'white','light red', 'white','#d66'
                 """.format(tag=tag)))
         self.addCleanup(os.unlink, f.name)
+        with self.assertRaises(re.error):
+            manager = SettingsManager()
+            manager.read_config(f.name)
+            manager.get_tagstring_representation(tag)
+
+    def test_translate_tagstring_prefix(self):
+        # Test for behavior mentioned in bcb2670f56fa251c0f1624822928d664f6455902,
+        # namely that 'foo' does not match 'foobar'
+        tag = 'foobar'
+        tagprefix = 'foo'
+        with tempfile.NamedTemporaryFile(mode='w+', delete=False) as f:
+            f.write(textwrap.dedent("""\
+                [tags]
+                    [[{tag}]]
+                        translated = matched
+                """.format(tag=tagprefix)))
+        self.addCleanup(os.unlink, f.name)
         manager = SettingsManager()
         manager.read_config(f.name)
-        manager.get_tagstring_representation(tag)
+        tagrep = manager.get_tagstring_representation(tag)
+        self.assertIs(tagrep['translated'], tag)
+        tagprefixrep = manager.get_tagstring_representation(tagprefix)
+        self.assertEqual(tagprefixrep['translated'], 'matched')
+
+    def test_translate_tagstring_prefix_regex(self):
+        # Test for behavior mentioned in bcb2670f56fa251c0f1624822928d664f6455902,
+        # namely that 'foo.*' does match 'foobar'
+        tagprefixregexp = 'foo.*'
+        with tempfile.NamedTemporaryFile(mode='w+', delete=False) as f:
+            f.write(textwrap.dedent("""\
+                [tags]
+                    [[{tag}]]
+                        translated = matched
+                """.format(tag=tagprefixregexp)))
+        self.addCleanup(os.unlink, f.name)
+        manager = SettingsManager()
+        manager.read_config(f.name)
+        translate = lambda t: manager.get_tagstring_representation(t)['translated']
+        matched = lambda t: translate(t) == 'matched'
+        self.assertTrue(all(map(matched, ['foo' 'foobar', tagprefixregexp])))
+        self.assertFalse(any(map(matched, ['bar', 'barfoobar'])))
+
+    def test_translate_regexp(self):
+        # Test for behavior mentioned in 108df3df8571aea2164a5d3fc42655ac2bd06c17
+        # namely that translations themselves can use regex
+        tag = "notmuch::foo"
+        section = "[[notmuch::.*]]"
+        translation = "'notmuch::(.*)', 'nm:\\1'" # actual config file doesn't need \\
```

Comment:
you can use raw strings to prevent double backslashing: `translation = r"'notmuch::(.*)', 'nm:\1'"`. Then you can leave out the comment.